src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -1097,20 +1127,22 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
         distinct = true;
       }
     }
+    ExpressionPtr result;
```

Comment:
Also remove the code duplication here:
```
auto makePtr = [&]<typename ExpressionType>(auto&&... additionalArgs) {
     ExpressionPtr result{std::make_unique<ExpressionType>(distinct, std::move(childExpression), AD_FWD(additionalArgs)...};
    result->descriptor() = context->getText();
   return result;
};
```
Example usage:
```
// if this is a count expression
return makePtr.operator()<CountExpression>(); // one line per case.
```
The additionalArgs come in handy for the group_cocat, which needs an additional argument.